pkg/test/cmp.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2019 The GitLab-Controller Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"reflect"
+
+	xpcorev1alpha1 "github.com/crossplaneio/crossplane/pkg/apis/core/v1alpha1"
+	"github.com/google/go-cmp/cmp"
+)
+
+// TODO(negz): Replace this if a similar option is added to cmpopts per
+// https://github.com/google/go-cmp/issues/89
+
+// EquateErrors returns true if the supplied errors are of the same type and
+// produce identical strings. This mirrors the error comparison behaviour of
+// https://github.com/go-test/deep, which most Crossplane tests targeted before
+// we switched to go-cmp.
+func EquateErrors() cmp.Option {
+	return cmp.Comparer(func(a, b error) bool {
+		if a == nil || b == nil {
```

Comment:
nit: consider 
```suggestion
		if a == nil {
		  return b == nil
		}
```

Reply:
Huh, I didn't know you could make suggestions like this. That's handy.

That said, are you sure that's the same logic? We would skip that branch if `b` was `nil` and `a` was not; I believe `cmp` Comparers must return the same result regardless of the order in which the two things being compared are provided, so we can't rely for example on `a` being the value we want.